Revamped applicant consent page

jeffmaher opened the pull request:
# Description

Implements two Trello tickets [(1)](https://trello.com/c/VEdSyzfN/416-applicants-can-give-or-not-give-consent-for-service-canada-to-obtain-their-personal-information-and-understands-what-that-means) [(2)](https://trello.com/c/fQSIlEx6/466-applicants-can-choose-who-they-give-consent-to).

Updates the applicant's consent page with:

- New content and visual design
- Ability to opt-out of data sharing with certain parties
- Ability to draw a signature
- Removed witness section
- No invite code shared if consent is declined (see notes)

It also adds the following things:

- Saving signature drawing and new opt-out fields
- Updates the view consent pages to display preferences and signature drawing

# Test

To test the applicant portion:

- Start a new application
- Fill out personal details
- Go to consent page
- Fill it out with a "yes" 
- Repeat and do with a "no"

To test doctor's view of the consent page:

- Do the above steps ^
- Login with invite code and birthdate
- Click "View Consent" on dashboard

To test MA view, do:

- Do the above ^ and complete an entire application
- Go to /en/sessions
- Click on your completed application

# Notes

- Not yet translated to French
- There is a [Trello card](https://trello.com/c/nWNylBaD/543-complete-medical-reports-where-the-applicant-decides-to-not-give-consent) that might change the behavior of not consenting in the future, to produce an invite code anyway. This PR implements the current design spec which calls for a completion page.

Commit:
Renders unstyled signature pad

Commit:
Functional signature pad

Commit:
Fixed size on signature box

Commit:
Re-aligned clear button

Commit:
Resize signature box width

Commit:
Get clear button to work

Commit:
Fix for the checkbox

Commit:
Updated content for applicant consent page

Commit:
Moved up signature line

Commit:
Remove medical consent macro

Commit:
Update styling for signature pad

Commit:
More removal of medical consent macro

Commit:
Package jSignature differently

Commit:
Ignore linting for jSignature

Commit:
Add safe attribute for radio buttons

Commit:
Show/hide signature pad

Commit:
Added warning text

Commit:
Updated some text capitalization

Commit:
Merge branch 'master' into signature_consent

Commit:
Remove whitespace

Commit:
Code cleanup on consent page

jeffmaher commented:
@cds-snc/esdc-cppd-design : This feature is not yet done, but could use some review on the visual and content design for the following things:

- Text content (made a few changes)
- Background color (which color from here best matches the design: https://tailwindcss.com/docs/background-color )
- How it looks on a phone (I'm particularly worried at how much text is jammed into the yes/no area)
- The draw versus type selector

Commit:
Add drawing data to hidden input field

Commit:
Added validation messages

Commit:
Added new database fields

Commit:
Data saving for consent form is in place

Commit:
Fields re-populate after error submissions

Commit:
Strip out SVG data from signature drawing

Commit:
Fix comparison error

Commit:
Updated invite page for no consent workflow

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.financial": "Financial institutions (for address updates only)",
+	"consent.volunteer": "Volunteer organizations",
+	"consent.employees": "Employees (for cases of self-employed persons)",
+	"consent.optional_parties_preamble": "Service Canada may also need to contact the organizations and persons listed below. Deselect any organizations and persons that Service Canada may not contact",
```
There's a period/full stop missing from the end of this string.

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.volunteer": "Volunteer organizations",
+	"consent.employees": "Employees (for cases of self-employed persons)",
+	"consent.optional_parties_preamble": "Service Canada may also need to contact the organizations and persons listed below. Deselect any organizations and persons that Service Canada may not contact",
+	"consent.question": "Do you give your consent to Service Canada to obtain medical and other personal informtion about you from the persons and organizations listed above?",
```
Typo in 'information'

jeffmaher replied:
Typo fixed. Ignoring required field since it's a bunch of checkboxes (chatted with Keith about it).

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
-	"section2.title": "Consent for Service Canada to obtain personal information",
-	"section2.witness_legend": "To be completed by a witness only if the applicant signs with a mark (e.g. X)",
-	"section2.witness_preamble": "I have read the contents of this section to the applicant, who appeared to fully understand them and who made their mark in my presence.",
+	"consent.yes": "<strong>Yes</strong>. I give my consent</strong> to Service Canada to obtain medical and other personal information about me from all persons and organizations listed above. I understand that this information may help determine if I qualify or continue to qualify for CPP disability benefits.",
```
Should read: `<strong>Yes</strong>, I give my consent. I understand that this information may help determine if I qualify or continue to qualify for CPP disability benefits.`
It's a bit shorter so should help with the reduced width on mobile screens.

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.cancel_in_writing": "in writing to Service Canada.",
+	"consent.cancel_consent_link": "https://catalogue.servicecanada.gc.ca/content/EForms/en/ReturningtheForm/IspCpp.html",
+	"consent.i_understand_intro": "I understand that if I do not give my consent, Service Canada:",
+	"consent.i_understand_1": "will make a decision based on the available information on my file",
```
Missing semicolon on the end of this line (not my choice!)

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.cancel_consent_link": "https://catalogue.servicecanada.gc.ca/content/EForms/en/ReturningtheForm/IspCpp.html",
+	"consent.i_understand_intro": "I understand that if I do not give my consent, Service Canada:",
+	"consent.i_understand_1": "will make a decision based on the available information on my file",
+	"consent.i_understand_2": "may stop paying me the benefits if I am already receiving them",
```
This line should read: `may stop paying me the benefits if I am already receiving them; and`
Again, not my choice.

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.i_understand_intro": "I understand that if I do not give my consent, Service Canada:",
+	"consent.i_understand_1": "will make a decision based on the available information on my file",
+	"consent.i_understand_2": "may stop paying me the benefits if I am already receiving them",
+	"consent.i_understand_3": "can require that I provide the necessary information",
```
Missing a full stop/period on the end.

keithiopia commented on views/pages/consent.njk:
```diff
@@ -1,36 +1,171 @@
+            ) 
+        }}
+
+        <div class="border border-gray-600 border-solid bg-gray-100 p-8">
```
Let's stick with `bg-gray-100`. It gives us AA with the red on gray, very close to AAA. Why are all the tailwind colours bluey-gray??

keithiopia commented on views/pages/consent.njk:
```diff
@@ -1,36 +1,171 @@
+                {% endif %}
+                <div id="signature_draw" class="relative w-3-4 border-2 border-black">
+                    <div id="signature_pad_clear" class="bottom-0 w-full text-right pr-4 pt-4 leading-none bg-white">
+                        <a onclick="$('#signature_pad').jSignature('clear')" class="no-underline text-gray-500">Clear</a>
```
Sadly the 'Clear' text fails contrast checks. We'll have to use `text-gray-700`.

keithiopia reviewed (commented):
Is it possible to increase the page width of the consent and signature box for mobile/cellphones? There's whitespace we can do without on the screen.
Oh, and can the CTA button say 'Submit' instead of 'Continue' ?

Commit:
Updated show consent form for 'yes' consent

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.yes": "<strong>Yes</strong>. I give my consent</strong> to Service Canada to obtain medical and other personal information about me from all persons and organizations listed above. I understand that this information may help determine if I qualify or continue to qualify for CPP disability benefits.",
+	"consent.no": "<strong>No</strong>. I do not give my consent",
+	"consent.title": "Consent for Service Canada to obtain personal information",
+	"consent.signature_draw": "Draw",
```
Can we change this to `Draw it` ?

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.no": "<strong>No</strong>. I do not give my consent",
+	"consent.title": "Consent for Service Canada to obtain personal information",
+	"consent.signature_draw": "Draw",
+	"consent.signature_type": "Type",
```
And this one to `Type it` ?

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.title": "Consent for Service Canada to obtain personal information",
+	"consent.signature_draw": "Draw",
+	"consent.signature_type": "Type",
+	"consent.signature_option": "How would you like to sign your signature?",
```
I'm going to say this should be "How would you like to give your signature?" merely because it is a bit of a tautology otherwise – sign and signature

keithiopia commented on config/locales/en.json:
```diff
@@ -313,24 +313,40 @@
+	"consent.signature_type": "Type",
+	"consent.signature_option": "How would you like to sign your signature?",
+	"consent.signature_draw_instructions": "Draw your signature with your mouse, touchpad, or finger.",
+	"consent.signature_type_instructions": "Type your full name with your keyboard.",
```
`Type your full name` should be sufficient here

Commit:
Added checklist for show consent page

Commit:
Switched to checkmarkList macro from checkmarkListItems

Commit:
Fixed invite screen and added submission date

Commit:
Added content and visual design fixes

Commit:
Changed submit button text

jeffmaher commented:
@keithiopia : Updated the button on the page to 'Submit'. I played around with the whitespace for a bit, but discovered that it's the margin that is used on all pages and that it's on every page and just not as noticeable since there are not many pages with highlighted sections like this. Could we fix the responsive design for all the pages in a future ticket, reducing the margins for mobile breakpoints globally?

Commit:
Fixed cancel button

Commit:
Merge branch 'master' into signature_consent

Commit:
Fixed linting issues

Commit:
Revert an accidental signature_type id

Commit:
Modularized the signature pad into a macro

Commit:
Made typed signatures bigger

jeffmaher commented on config/webpack.js:
```diff
@@ -23,6 +23,10 @@ module.exports.webpack = {
+          },
+          {
+            from: './assets/js/jSignature.min.js',
+            to: path.resolve(__dirname, '..', '.tmp', 'public', 'build', 'assets', 'js')
```
This library suggests pulling down the files for manual install, so this is why this is harder than using another JS library that either has an NPM package or has a reliable/known CDN.

jeffmaher commented on views/macros/medical-consent.njk:
```diff
@@ -1,29 +0,0 @@
-{% macro medicalConsent(key, values, value, question, errors, attributes) %}
```
No longer used, so deleted.

CalvinRodo replied:
Glad to see it gone!

jeffmaher commented on views/macros/radios.njk:
```diff
@@ -20,7 +20,13 @@
                         <input id="{{ key }}{{ val }}" name="{{ key }}" type="radio" value="{{ index }}" {% if value == index %} checked="checked" {% endif %} {% if errors and errors[key] %} aria-describedby="{{ key + '-error' }}" aria-invalid="true" {% endif %}>
-                        <label for="{{ key }}{{ val }}">{{ __(val) }}</label>
+                        <label for="{{ key }}{{ val }}">
+                            {% if attributes.safe %}
```
Added a `safe` attribute to handle translations with embedded formatting.

jeffmaher commented on views/macros/signaturePad.njk:
```diff
@@ -0,0 +1,103 @@
+{#
```
Macro-ized the signature pad, so that we can use it again for the medical professional consent.

cc @dsamojlenko

dsamojlenko replied:
beauty

CalvinRodo commented:
When the cursor is over __clear__ it renders an icon for select text, since it's just a link should the hover icon be the same as when selecting a button?

CalvinRodo commented:
Show Consent link throws a 500 error on the Medical Adjuticator view with the following exception getting thrown

`Error: WHERE parameter "applicationCode" has invalid "undefined" value`

keithiopia commented:
> @keithiopia : Updated the button on the page to 'Submit'. I played around with the whitespace for a bit, but discovered that it's the margin that is used on all pages and that it's on every page and just not as noticeable since there are not many pages with highlighted sections like this. Could we fix the responsive design for all the pages in a future ticket, reducing the margins for mobile breakpoints globally?

Yeah, let's make a ticket for that. Given how dense some of the text we have to use is, this will be a problem on other pages too.

kylejones-ESDC reviewed (commented):
Print this code - Print view shows phrase "Print this code" on printable document.
MP view - When clicking "View consent" I receive a 500 error.
- When I click any of the links (except functional limitations and restrictions) in the MP view I am returned back to the applicant view where they request they enter their information to obtain a code.
- when I click functional limitations I get a 500 error code.
Unable to reach the MA view.

shannd reviewed (changes requested):
Applicant Portion
- I get a lot of error messages with the SIN, I had to try 6 or more number combinations for Mary Poppins 

Doctors view
- The time stamp for when the consent submitted was correct in terms of date (Feb. 6th) but not time (15:05) 
- I have content comments: will a doctor know what we mean by "file development"
- Should we also say that consent is for the purposes of determining eligibility or continuing eligibility? 

Filling out the MR:
- Under Basic Information the MR asks "did you recommend they stop working", I choose no. When I tried to save this section I got an error message that said the "when did you recommend they stop working" field was required. This required field doesn't show up unless you choose "yes".

keithiopia commented:
> Doctors view
> * I have content comments: will a doctor know what we mean by "file development"

@shannd The doctors shouldn't be seeing anything about 'file development'. Which page was that on?

shannd commented:
> > Doctors view
> > 
> > * I have content comments: will a doctor know what we mean by "file development"
> 
> @shannd The doctors shouldn't be seeing anything about 'file development'. Which page was that on?

@keithiopia It was on the page where the doctor can view consent. The top of the page reads "consent for file development" or something like that.

keithiopia commented:
> > > Doctors view
> > > 
> > > * I have content comments: will a doctor know what we mean by "file development"
> > 
> > 
> > @shannd The doctors shouldn't be seeing anything about 'file development'. Which page was that on?
> 
> @keithiopia It was on the page where the doctor can view consent. The top of the page reads "consent for file development" or something like that.

@shannd Ahh, that sounds like we're showing doctors the MA consent download. Good spot!

shannd commented:
> > > > Doctors view
> > > > 
> > > > * I have content comments: will a doctor know what we mean by "file development"
> > > 
> > > 
> > > @shannd The doctors shouldn't be seeing anything about 'file development'. Which page was that on?
> > 
> > 
> > @keithiopia It was on the page where the doctor can view consent. The top of the page reads "consent for file development" or something like that.
> 
> @shannd Ahh, that sounds like we're showing doctors the MA consent download. Good spot!

@keithiopia I also think that page (where the doctor can view consent) should say more of what the consent is for, i.e. to determine eligibility or continuing eligibility for CPP disability benefits. While we're saying they've consented to reach out to these people, it's not clear what that is for.

jeffmaher commented:
Thanks for the feedback! Jumping in now to fix stuff. Will ping back here with questions/clarifications.

dsamojlenko committed:
Merge branch 'master' into signature_consent

jeffmaher commented:
> Show Consent link throws a 500 error on the Medical Adjuticator view with the following exception getting thrown
> 
> `Error: WHERE parameter "applicationCode" has invalid "undefined" value`

@CalvinRodo : Can you send me thorough steps to recreate? I'm having trouble getting this error to happen (I'm testing via the Heroku deployment from `/en/sessions`).

dsamojlenko commented:
> When the cursor is over **clear** it renders an icon for select text, since it's just a link should the hover icon be the same as when selecting a button?

You should be able to just add `cursor-pointer` class to that element, ie: 
![image](https://user-images.githubusercontent.com/1187115/73965686-9b16e880-48e2-11ea-9334-a9220e647e45.png)

dsamojlenko commented:
> > Show Consent link throws a 500 error on the Medical Adjuticator view with the following exception getting thrown
> > `Error: WHERE parameter "applicationCode" has invalid "undefined" value`
> 
> @CalvinRodo : Can you send me thorough steps to recreate? I'm having trouble getting this error to happen (I'm testing via the Heroku deployment from `/en/sessions`).

That'll be because of this: 
![image](https://user-images.githubusercontent.com/1187115/73965866-eb8e4600-48e2-11ea-894f-d3431b53afe4.png)

... the medical report in this context should come from the URL not the session. See the sessions/view routes for an example.

keithiopia commented:
> Thanks for the feedback! Jumping in now to fix stuff. Will ping back here with questions/clarifications.

We've moved the view consent for doctors conversation elsewhere.

dsamojlenko commented on views/macros/ul-checkmark.njk:
```diff
@@ -0,0 +1,21 @@
+                {% else %}
+                    <svg class="inline h-5 w-5" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20"><path d="M14.348 14.849a1.2 1.2 0 0 1-1.697 0L10 11.819l-2.651 3.029a1.2 1.2 0 1 1-1.697-1.697l2.758-3.15-2.759-3.152a1.2 1.2 0 1 1 1.697-1.697L10 8.183l2.651-3.031a1.2 1.2 0 1 1 1.697 1.697l-2.758 3.152 2.758 3.15a1.2 1.2 0 0 1 0 1.698z"/></svg>
+                {% endif %}
+                &nbsp;{{ __(key) }}
```
instead of a nbsp, I think you could add a `whitespace-no-wrap` class to the parent element?

jeffmaher replied:
Will do!

jeffmaher replied:
Just removed the nbsp and it looks fine (even without the class). Good catch!

dsamojlenko replied:
I have a (probably undeserved) aversion to nbsp.

Commit:
Fixed cursor pointer bug on signature clear link

jeffmaher commented:
> > @keithiopia : Updated the button on the page to 'Submit'. I played around with the whitespace for a bit, but discovered that it's the margin that is used on all pages and that it's on every page and just not as noticeable since there are not many pages with highlighted sections like this. Could we fix the responsive design for all the pages in a future ticket, reducing the margins for mobile breakpoints globally?
> 
> Yeah, let's make a ticket for that. Given how dense some of the text we have to use is, this will be a problem on other pages too.

Done! https://trello.com/c/0sAz6ggj/704-reduce-border-whitespace-around-forms-on-smaller-displays

Commit:
Fixed 500 error when viewing consent as an MA

jeffmaher commented:
> Print this code - Print view shows phrase "Print this code" on printable document.
> MP view - When clicking "View consent" I receive a 500 error.
> 
>     * When I click any of the links (except functional limitations and restrictions) in the MP view I am returned back to the applicant view where they request they enter their information to obtain a code.
> 
>     * when I click functional limitations I get a 500 error code.
>       Unable to reach the MA view.

A few quick notes:

- Fixed the 500 error when trying to view consent
- Added [a Trello ticket](https://trello.com/c/PpMXZFPW/706-remove-print-button-from-invite-screen-on-printout) for hiding the print button (since it's a different scope)
- Couldn't recreate the issue with the MP view. Could you send step-by-step instructions on how you experienced the error? That'll help me recreate the conditions in which it occurred.

jeffmaher commented:
> Applicant Portion
> 
>     * I get a lot of error messages with the SIN, I had to try 6 or more number combinations for Mary Poppins
> 
> 
> Doctors view
> 
>     * The time stamp for when the consent submitted was correct in terms of date (Feb. 6th) but not time (15:05)
> 
>     * I have content comments: will a doctor know what we mean by "file development"
> 
>     * Should we also say that consent is for the purposes of determining eligibility or continuing eligibility?
> 
> 
> Filling out the MR:
> 
>     * Under Basic Information the MR asks "did you recommend they stop working", I choose no. When I tried to save this section I got an error message that said the "when did you recommend they stop working" field was required. This required field doesn't show up unless you choose "yes".

Thanks for the info @shannd ! Some quick notes:

- The SIN needs to be a valid SIN. There is algorithm in place that checks to see if it is a valid SIN, so that's likely the source of errors. I just use the example SIN for testing. If you think there's still an issue with it, can you file a bug Trello with some of the valid SINs you may have tried (since it's a different scope than this consent feature)?
- Good spot on the time! It's currently showing GMT, so it's currently timezone agnostic. I can fix this, but have a few questions for @keithiopia and @michellengaiCDS. Those are (A) How important is it to show the time on the consent form? (i.e. should we invest time into fixing this?) (B) Should it show the timezone of the applicant or the timezone of the viewer (i.e. the MP or MA)?
- Can you file a separate bug Trello card for the basic information issue, since that's a different scope than the consent form?

Commit:
Fixed title of viewing consent form

jeffmaher commented:
> > > > Doctors view
> > > > 
> > > > * I have content comments: will a doctor know what we mean by "file development"
> > > 
> > > 
> > > @shannd The doctors shouldn't be seeing anything about 'file development'. Which page was that on?
> > 
> > 
> > @keithiopia It was on the page where the doctor can view consent. The top of the page reads "consent for file development" or something like that.
> 
> @shannd Ahh, that sounds like we're showing doctors the MA consent download. Good spot!

Fixed!

jeffmaher commented on views/pages/consent.njk:
```diff
@@ -2,36 +2,85 @@
+                        </ul>
+                    </td>
+                    <tr/>
+            </table>
```
Thanks! Pulled in the changes.

Commit:
Replaced table on consent form for a11y

Commit:
Use fullName instead of split names

jeffmaher commented on api/controllers/ConsentController.js:
```diff
@@ -32,19 +51,34 @@ module.exports = {
+          consentFinancial: req.body.consent_optional_parties.includes('financial'),
+          consentVolunteer: req.body.consent_optional_parties.includes('volunteer'),
+          consentEmployees: req.body.consent_optional_parties.includes('employees'),
+          signatureMode: req.body.signatureMode,
```
Good point! Working on fixing now.

jeffmaher replied:
Fixed!

Commit:
Added removed whitespace

jeffmaher commented:
Keith and I chatted. We're going to remove the timestamp (keeping "month day, year") from the consent form, since it requires extra work to capture the applicant's timezone and that it's not an important piece of info to display.

Commit:
Removed use of DB signatureMode

jeffmaher commented:
I think I got to all the feedback that needed fixes. Ready for re-review.

dsamojlenko reviewed (dismissed):
Looks good, good work Jeff! 🌮

Commit:
Merge branch 'master' into signature_consent

keithiopia commented on config/locales/en.json:
```diff
@@ -305,24 +305,40 @@
+	"consent.government": "federal, provincial, territorial, or municipal government departments and agencies",
+	"consent.compensation_boards": "provincial or territorial workers' compensation boards",
+	"consent.employers": "employers, former employers",
+	"consent.insurers": "administators of insurance plans",
```
```suggestion
	"consent.insurers": "administrators of insurance plans",
```
Small typo

shannd reviewed (changes requested):
- filled out the entire MR report and went to submit. This message was displayed “This is going to be something else”
- went to the MA view and couldn't find the submitted MR for Shannon Doyle. There are three pages of submitted MR's but each page of names appears to be the same

keithiopia commented on config/locales/en.json:
```diff
@@ -305,24 +305,40 @@
-	"section2.witness_legend": "To be completed by a witness only if the applicant signs with a mark (e.g. X)",
-	"section2.witness_preamble": "I have read the contents of this section to the applicant, who appeared to fully understand them and who made their mark in my presence.",
+	"consent.yes": "<strong>Yes</strong>, I give my consent. I understand that this information may help determine if I qualify or continue to qualify for CPP disability benefits.",
+	"consent.no": "<strong>No</strong>. I do not give my consent",
```
```suggestion
	"consent.no": "<strong>No</strong>, I do not give my consent.",
```
Sorry, I should have caught this before. Comma instead of period.

keithiopia commented on config/locales/en.json:
```diff
@@ -479,15 +495,12 @@
-	"show_consent.patient_signature": "Patient signature",
-	"show_consent.patient_name": "Patient name",
-	"show_consent.agreement": "<p>Service Canada is authorized under Section 68 and 69 of <em>the Canada Pension Plan (CPP) Regulations</em> to receive personal information (medical and non-medical) about you to determine if you qualify or continue to qualify for CPP disability benefits. Your consent to permit Service Canada to obtain this information is necessary, should Service Canada need this in ...
+	"show_consent.title": "'s consent",
```
```suggestion
	"show_consent.title": "'s consent to share medical and personal information",
```
Just to match the Figma, and make it clear what the doc is

jeffmaher commented:
>     * filled out the entire MR report and went to submit. This message was displayed “This is going to be something else”
> 
>     * went to the MA view and couldn't find the submitted MR for Shannon Doyle. There are three pages of submitted MR's but each page of names appears to be the same

@shannd :

- For the final submission part, can you submit a bug for that. It's a little outside the scope.
- For the MA view, the pagination of the list is a separate scope, but def makes it hard to test. I played with the URL to see if I could find your submission and found this: https://cppd-medical-signature--nvexb3.herokuapp.com/en/sessions/18/view . Here's some screenshots:

![image](https://user-images.githubusercontent.com/950486/74055251-a6831600-49ad-11ea-9b53-e5ab7b72d22f.png)

![image](https://user-images.githubusercontent.com/950486/74055273-b1d64180-49ad-11ea-9664-b82b9b5cf897.png)

Do those look like what you're expecting? (I'd encourage you to try the link from an Incognito tab too, using the link above)

shannd commented:
> > ```
> > * filled out the entire MR report and went to submit. This message was displayed “This is going to be something else”
> > 
> > * went to the MA view and couldn't find the submitted MR for Shannon Doyle. There are three pages of submitted MR's but each page of names appears to be the same
> > ```
> 
> @shannd :
> 
> * For the final submission part, can you submit a bug for that. It's a little outside the scope.
> * For the MA view, the pagination of the list is a separate scope, but def makes it hard to test. I played with the URL to see if I could find your submission and found this: https://cppd-medical-signature--nvexb3.herokuapp.com/en/sessions/18/view . Here's some screenshots:
> 
> ![image](https://user-images.githubusercontent.com/950486/74055251-a6831600-49ad-11ea-9b53-e5ab7b72d22f.png)
> 
> ![image](https://user-images.githubusercontent.com/950486/74055273-b1d64180-49ad-11ea-9664-b82b9b5cf897.png)
> 
> Do those look like what you're expecting? (I'd encourage you to try the link from an Incognito tab too, using the link above)

Yup, that looks right! Will file a ticket for the other one and work on my scope creep :(

shannd reviewed (dismissed):
Looks good :)

keithiopia commented on config/locales/en.json:
```diff
@@ -305,24 +305,40 @@
+	"consent.signature_draw": "Draw it",
+	"consent.signature_type": "Type it",
+	"consent.signature_option": "How would you like to give your signature?",
+	"consent.signature_draw_instructions": "Draw your signature with your mouse, touchpad, or finger.",
```
```suggestion
	"consent.signature_draw_instructions": "Draw your signature with your mouse, touchpad, or screen.",
```
This way the list is about the tech rather than technique/digit.

jeffmaher committed:
Update config/locales/en.json

Co-Authored-By: Keith Emmerson <keith.emmerson@cds-snc.ca>

jeffmaher committed:
Update config/locales/en.json

Co-Authored-By: Keith Emmerson <keith.emmerson@cds-snc.ca>

jeffmaher committed:
Update config/locales/en.json

Co-Authored-By: Keith Emmerson <keith.emmerson@cds-snc.ca>

jeffmaher committed:
Update config/locales/en.json

Co-Authored-By: Keith Emmerson <keith.emmerson@cds-snc.ca>

jeffmaher commented:
@keithiopia : Thanks for the suggested commits! Super easy to get them in. Those are all in now.

dsamojlenko committed:
remove empty

dsamojlenko reviewed (approved):
LGTM 💰

keithiopia commented:
In terms of scope, I assume we're just testing the functionality of the consent download artefacts, rather than the contents of them?
